xbmc/filesystem/File.cpp

```diff
@@ -243,10 +243,7 @@ bool CFile::CURLAddOption(XFILE::CURLOPTIONTYPE type, const char* name, const ch
 
 bool CFile::CURLOpen(unsigned int flags)
 {
-  if (flags & XFILE::READ_AFTER_WRITE)
-    return OpenForWrite(m_curl, true);
-  else
-    return Open(m_curl, flags);
+  return Open(m_curl, flags);
```

Comment:
The flag was introduced in one of my PR's some weeks ago. It was necessary to wait for the CURL action because writing of POSTDATA has to wait.
This is obsolete now: PostData is prepared before any open call is done.
I also removed the XFILE::READ_AFTER_WRITE flag. No need for this anymore.